ui/src/app/pages/workspace/create-billing-account-modal.spec.tsx

```diff
@@ -0,0 +1,95 @@
+import {mount} from 'enzyme';
+import * as React from 'react';
+
+import {registerApiClient} from 'app/services/swagger-fetch-clients';
+import {WorkspaceAccessLevel, WorkspacesApi} from 'generated/fetch';
+import {waitOneTickAndUpdate} from 'testing/react-test-helpers';
+import {workspaceStubs} from 'testing/stubs/workspaces';
+import {WorkspacesApiStub} from 'testing/stubs/workspaces-api-stub';
+import {WorkspaceCard} from './workspace-card';
+import {profileStore} from 'app/utils/stores';
+import defaultServerConfig from 'testing/default-server-config';
+import {CreateBillingAccountModal} from "./create-billing-account-modal";
+import {ProfileStubVariables} from 'testing/stubs/profile-api-stub';
+
+describe('CreateBillingAccountModal', () => {
+  const load = jest.fn();
+  const reload = jest.fn();
+  const updateCache = jest.fn();
+  const profile = ProfileStubVariables.PROFILE_STUB;
+
+  const component = () => {
+    return mount(<CreateBillingAccountModal onClose={() => {}}/>);
+  };
+
+  beforeEach(() => {
+    profileStore.set({profile, load, reload, updateCache});
+  });
+
+  it('simulate creating billing account with all steps correctly', async() => {
```

Comment:
this test is handling a lot, generally easier to debug and understand unit test failures if you split out into smaller and more targeted test-cases